### PR TITLE
Annalyn's Infiltration  - Task 4 - Instructions don't match tests

### PR DESCRIPTION
**Pull request by sarad1p1ty**
As per #758:

- [X] Add a logic table (inputs and expected result) to exercises/concept/annalyns-infiltration/.meta/design.md to ensure proper wording of instructions for the future
~~- [ ] Find wording that matches the logic of the tests~~
- [ ] Change Instructions wording to explicitly state that if the dog is present it does not matter if the prisoner is asleep or awake.
- [X] Order tests to follow the sequence of instruction conditions (e.g. the first statement is "If Annalyn has her pet dog with her" -> first all tests "with dog" then all tests "without dog")
- [x] Ask in forum about wording and logic issues of the new version
- [X] Rename exercise title to Annalyn's Infiltration (apostrophe added)

### CODE REVIEW

**Comment by mk-mxp** [comment]
I think you can't look into the CI details, there are some formatting issues reported:

```
> phpcs

FILE: ...p/php/exercises/concept/annalyns-infiltration/AnnalynsInfiltrationTest.php
--------------------------------------------------------------------------------
FOUND 8 ERRORS AFFECTING 8 LINES
--------------------------------------------------------------------------------
 223 | ERROR | [x] Whitespace found at end of line
     |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
 240 | ERROR | [x] Whitespace found at end of line
     |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
 257 | ERROR | [x] Whitespace found at end of line
     |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
 274 | ERROR | [x] Whitespace found at end of line
     |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
 308 | ERROR | [x] Whitespace found at end of line
     |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
 325 | ERROR | [x] Whitespace found at end of line
     |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
 410 | ERROR | [x] Whitespace found at end of line
     |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
 461 | ERROR | [x] Whitespace found at end of line
     |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
```

File: exercises/concept/annalyns-infiltration/AnnalynsInfiltrationTest.php [hunk 1]
```diff
@@ -80,37 +80,38 @@ public function testCanSpyWhenOnlyArcherAwake()
...
-            is_knight_awake: true,
-            is_archer_awake: false,
-            is_prisoner_awake: false
+            is_knight_awake: false,
+            is_archer_awake: true,
+            is_prisoner_awake: true
         );
         $expected = true;
         $this->assertEquals($expected, $actual);
     }
```
Comment (mk-mxp) [inline exercises/concept/annalyns-infiltration/AnnalynsInfiltrationTest.php:112]: One test case is missing to cover all combinations of Task 2:

```php
    /**
     * @testdox can spy when only the prisoner is asleep
     * @task_id 2
     */
    public function testCanSpyWhenOnlyPrisonerAsleep()
    {
        $infiltration = new AnnalynsInfiltration();
        $actual = $infiltration->canSpy(
            is_knight_awake: true,
            is_archer_awake: true,
            is_prisoner_awake: false
        );
        $expected = true;
        $this->assertEquals($expected, $actual);
    }
```

**Review by mk-mxp — CHANGES_REQUESTED** [review]
Thanks for investing your time into this! Looking forward to merging soon.

File: config.json [hunk 2]
```diff
@@ -1508,4 +1508,4 @@
     "used_for/scripts",
     "used_for/web_development"
   ]
-}
+}
```
[the remark this answers is not in the document]
Comment (sarad1p1ty) [reply config.json:1511]: Thank you.

I'm afraid I'm having a little trouble with the CI tools. I am using VS Code locally, and thought I followed all instructions for installing configlet, but when I run `bin/configlet fmt` (or `composer run ci`), I am getting `Script bin/configlet fmt handling the configlet:fmt event returned with error code 1`

I'm not sure what I'm doing wrong.

File: exercises/concept/annalyns-infiltration/AnnalynsInfiltrationTest.php (hunk 1 above)
Comment (sarad1p1ty) [reply exercises/concept/annalyns-infiltration/AnnalynsInfiltrationTest.php:112]: Good spot! I have added this.

File: config.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (mk-mxp) [reply config.json:1511]: Troubleshooting external tools is so much fun! You are on a Linux machine or Windows, Apple, ...? `configlet` is not compiled for all architectures / systems, did `bin/fetch-configlet` actually download a binary to `bin/`?

[the remark this answers is not in the document]
Comment (sarad1p1ty) [reply config.json:1511]: Ah... oh dear.

Right, so I'm on a Windows 10 machine. So `fetch-configlet` seems to have downloaded `configlet.exe` to `bin`, yes.

[the remark this answers is not in the document]
Comment (mk-mxp) [reply config.json:1511]: Great! Runniing `bin/configlet.exe fmt` works`? Editing `composer.json` to use `bin/configlet.exe` works?

[the remark this answers is not in the document]
Comment (mk-mxp) [reply config.json]: Great you found your way to fix this! CI is pleased :smiley:

[the remark this answers is not in the document]
Comment (sarad1p1ty) [reply config.json]: Looks like this works now?

[the remark this answers is not in the document]
Comment (mk-mxp) [reply config.json]: Yes, it works. Do you want to take an appraoch with the instructions or should I merge as it is? I'm OK with both.

[the remark this answers is not in the document]
Comment (sarad1p1ty) [reply config.json]: Exciting!

If you're ok with merging as is, then that might be a good shout. "If it ain't broke..." :)

Thank you.

**Review by mk-mxp — APPROVED** [review]
Thanks for your contribution! There where no suggestions for wording changes in the forum, too.

**Comment by sarad1p1ty** [comment]
Thank you! :D 
Is there anything else I can do?

**Comment by mk-mxp** [comment]
@sarad1p1ty Did you look through the GitHub issues list? There is lots of things to do, with and without in-depth PHP / PHPUnit / Exercism knowledge. Synchronising exercises from the [common problem specifications repository](https://github.com/exercism/problem-specifications), adding missing exercises from that repository, adding concept exercises (adapted from other Exercism languages with similar concepts) aso. Pick something and I'll make an issue / forum thread with detailed instructions.

**Pull request merged**